Setup/env_test.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/sudo /bin/bash
+
+LOGFILE="${LOGFILE:-./env_test.log}"
+URLS_TO_CHECK='http://www.google.com/ https://www.google.com/ http://www.ericom.com/ https://www.ericom.com/ https://hub.docker.com/'
+SHIELD_NETWORK_ADDR_BLOCK='10.20.0.0/16'
+
+if ! declare -f log_message >/dev/null; then
+    function log_message() {
+        echo "$1"
+        echo "$(date): $1" >>"$LOGFILE"
+    }
+fi
+
+if ! declare -f failed_to_install >/dev/null; then
+    function failed_to_install() {
+        log_message "An error occured during the installation: $1, Exiting!"
+        exit 1
+    }
+fi
+
+function check_url_connectivity() {
+    printf "\nChecking $1 ..."
+    if ! curl "$1" -sS -o /dev/null -w "\nResponse Code: %{http_code}\nDNS time: %{time_namelookup}\nConnection time: %{time_connect}\nPretransfer time: %{time_pretransfer}\nStarttransfer time: %{time_starttransfer}\nTotal time: %{time_total}\n"; then
+        printf "$1 check failed"
+        return 1
+    fi
+}
+
+function check_connectivity() {
+    for url in $URLS_TO_CHECK; do
+        if ! check_url_connectivity "$url"; then
+            echo "Connectivity test failed for $url"
+            return 1
+        fi
+    done
+}
+
+function check_storage_drive_speed() {
+    /sbin/hdparm -Tt $(df -l --output=source /var/lib | awk 'FNR == 2 {print $1}')
```

Comment:
what is the pass/fail criteria here?

Reply:
It is only a performance test, it gives numbers and should not fail.